workflow/controller/testdata/workflow-sub-test-6.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: workflow-template-hello-world-
+  namespace: test
+  labels:
+    mutexName: myMutex
+  annotations:
+    schedulerName: myScheduler
+spec:
+  workflowTemplateRef:
+    name: workflow-template-submittable
+  synchronization:
+    mutex:
+      name:  "{{workflow.labels.mutexName}}"
+  schedulerName:  "{{workflow.annotations.schedulerName}}"
+  workflowMetadata:
+    labelsFrom:
+      mutexName: 
+        expression: "{{= sprig.quote(sprig.trim( workflow.annotations.schedulerName )) }}"
+    annotations:
+      schedulerName: wfMetadataScheduler
```

Comment:
In this new test, we verify that:

1. workflowMetadata.labelsFrom.mutexName.expression (which is based on another parameter) gets evaluated correctly
2. the result of that label gets incorporated into spec.synchronization.mutex.name